src/main/java/be/seeseemelk/mockbukkit/entity/EntityEquipmentMock.java

```diff
@@ -231,8 +240,12 @@ public void setBoots(ItemStack boots, boolean silent)
 	@Override
 	public void setArmorContents(ItemStack[] items)
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		if (items.length == 4) {
+			setHelmet(items[0]);
+			setChestplate(items[1]);
+			setLeggings(items[2]);
+			setBoots(items[3]);
+		}
```

Comment:
else?

Reply:
We don't know what to do if we give a different array size. Should we add the first 4 items ? Should we do nothing ? Should we throw an Exception ?

Reply:
We certainly know what to do.
MockBukkit is a **mock** of CraftBukkit after all, so all methods should always reflect the implementations found in CraftBukkit, for this method the standard implementation is as follows:

```java
    @Override
    public void setArmorContents(final ItemStack[] items) {
        this.setEquipment(EnumItemSlot.FEET, (items.length >= 1) ? items[0] : null);
        this.setEquipment(EnumItemSlot.LEGS, (items.length >= 2) ? items[1] : null);
        this.setEquipment(EnumItemSlot.CHEST, (items.length >= 3) ? items[2] : null);
        this.setEquipment(EnumItemSlot.HEAD, (items.length >= 4) ? items[3] : null);
    }
```

Not only did you get the order completely reversed, but smaller arrays are completely ignored, whereas they should be allowed.